### ISSUE TITLE
Siblings aggregation on vectors issue with multi level hierarchies 

### ISSUE THREAD

**Issue by SamTaz** [atoti/atoti#16]
Hello,

I'm working on a VaR notebook ([you can get it gere, it downloads the necessary data itself](https://github.com/atoti/notebooks/blob/sta/var/finance/value-at-risk/main.ipynb)) and I seem to have an issue with sibling aggregation on the VaR in a multi-level hierarchy: instead of aggregating only on the siblings that share the same parent level, it seems the sibling aggregation is performed on all members.

![image](https://user-images.githubusercontent.com/39309622/79338262-47023300-7f27-11ea-8a32-9e48e6734f31.png)

Where the VaR **(values ok)** is defined by:
```
scaled_pnl_vector = m["Quantity"] * m["pnl_vector.VALUE"]
m['Position Vector'] = atoti.agg.sum(scaled_pnl_vector, scope=atoti.scope.origin('instrument_code'))
m["Confidence Level"] = 0.95
m["VaR"] = atoti.array.percentile(m["Position Vector"], m["Confidence Level"])
```

Parent VaR **(values ok)**:
`m["Parent VaR"] = atoti.parent_value(m["VaR"], h["Trading Book Hierarchy"])`

VaR agg on siblings **(values not ok)**:
```
m["Position Vector Agg on siblings"] = atoti.agg.sum(
    m["Position Vector"],
    scope=atoti.scope.siblings(h["Trading Book Hierarchy"]),
)
m["VaR agg on siblings"] = atoti.array.percentile(m["Position Vector Agg on siblings"], m["Confidence Level"])
```

Parent VaR Ex **(values not ok)**:
```
m["Parent Position Vector Ex"] = atoti.agg.sum(
    m["Position Vector"],
    scope=atoti.scope.siblings(h["Trading Book Hierarchy"], exclude_self=True),
)
m["Parent VaR Ex"] = atoti.array.percentile(m["Parent Position Vector Ex"], m["Confidence Level"])
```

And in the end the marginal VaR (values not ok):
`m["Marginal VaR"] = m["Parent VaR"] - m["Parent VaR Ex"]`

In this particular example, the value of the `Parent VaR Ex` at level `Hermes` should be equal to the value of the `VaR` at level `Developed Market` after having applied a filter excluding `Hermes`.

Thanks in advance for the help.

PS : I don't think it is related to [this issue](https://github.com/activeviam/atoti/issues/1058) as I also tried it with `cube.aggregates_cache.capacity = 0`

**Comment by APolyakova** [comment]
Can not reproduce this issue on 4.2+. Seems to be fixed. Please confirm

**Comment by a-manai** [comment]
Hello,

This issue has been fixed in 0.4.1.

Cheers
